audit-service/src/main/java/uk/gov/gchq/palisade/audit/service/request/AuditRequest.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.gchq.palisade.audit.service.request;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import uk.gov.gchq.palisade.RequestId;
+import uk.gov.gchq.palisade.service.request.Request;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Date;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This is the abstract class that is passed to the audit-service
+ * to be able to store an audit record. The default information is
+ * when was the audit record created and by what server.
+ * <p>
+ * The four immutable data subclasses below can be instantiated by static
+ * {@code create(RequestId orig)} factory methods which chain construction by fluid interface definitions.
+ */
+
+@JsonTypeInfo(
+        use = JsonTypeInfo.Id.NAME,
+        include = JsonTypeInfo.As.PROPERTY,
+        property = "class"
+)
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = RegisterRequestCompleteAuditRequest.class),
+        @JsonSubTypes.Type(value = RegisterRequestExceptionAuditRequest.class),
+        @JsonSubTypes.Type(value = ReadRequestCompleteAuditRequest.class),
+        @JsonSubTypes.Type(value = ReadRequestExceptionAuditRequest.class)
+})
+public class AuditRequest extends Request {
+
+    public final Date timestamp;
```

Comment:
The Date class is very very legacy. Please could this be changed to something from the java.time API like java.time.ZonedDateTime?

Reply:
```suggestion
    public final ZoneDateTime timestamp;
```